plugins/m/matplotlib_figure.py

```diff
@@ -67,7 +67,9 @@ def run_after_snippet(self, gl):
         os.makedirs(os.path.dirname(output_img_filename), exist_ok=True)
 
         # store relative path
-        self.image_uri = os.path.relpath(output_img_filename, os.path.dirname(source_filename))
+        # TODO: need a reliably way to find resulting .html path, now it's assumed to be output root
+        self.image_uri = os.path.relpath(output_img_filename, self.input_root)
```

Comment:
I don't want to scatter generated images across source tree, so I collect everything in `$SOURCE/matplotlib-figures/` directory. Not a bad idea, right? This creates a problem. 

Relative path to image is different in `SOURCE` (relative to .rst) and `OUTPUT` (relative to .html) directories now, since all generated `.html` are piled in one directory (e.g. `$SOURCE/bar/baz/page.rst` is rendered as `$OUTPUT/page.html`). 

Pelican doesn't care much about linked images, so I can simply use relative path in output directory as image source.

mcss does some pre-processing & auto copy of linked images to output directory. This creates a problem because mcss operates on relative paths in `$SOURCE` directory with assumption they are same as in output and fails.

Should `$SOURCE/bar/baz/page.rst` - like layouts be marked as "unsupported" or there should be a way to handle it in clever way?